File: common/src/main/java/vip/qoriginal/quantumplugin/registration/MinecraftRegistrationTest.java
package vip.qoriginal.quantumplugin.registration;

import org.bukkit.entity.Player;

public interface MinecraftRegistrationTest {
    String METHOD_ID = "minecraft";
    String SESSION_ENDPOINT = "/qo/registration/minecraft/session";
    String CLAIM_ENDPOINT = "/qo/registration/minecraft/claim";
    String RESULT_ENDPOINT = "/qo/registration/minecraft/result";

    boolean isAvailable();

    /**
     * Starts a test for an online player. Implementations and callers must invoke this on the
     * Minecraft server thread because a test may teleport the player immediately.
     */
    StartResult start(Player player, Session session);

    void cancel(Player player);

    record Session(String sessionId, String username) {
    }

    record StartResult(boolean accepted, String code, String message) {
        public static StartResult unavailable() {
            return new StartResult(
                    false,
                    "minecraft_verification_reserved",
                    "Minecraft 世界测试接口已预留，当前尚未开放。"
            );
        }
    }
}

File: common/src/main/java/vip/qoriginal/quantumplugin/registration/MinecraftRegistrationTest.kt
package vip.qoriginal.quantumplugin.registration

import org.bukkit.entity.Player

interface MinecraftRegistrationTest {
    fun isAvailable(): Boolean

    /**
     * Starts a test for an online player. Implementations and callers must
     * invoke this on the Minecraft server thread because a test may teleport
     * the player immediately.
     */
    fun start(player: Player, session: Session): StartResult

    fun cancel(player: Player)

    @JvmRecord
    data class Session(
        val sessionId: String,
        val username: String,
    )

    @JvmRecord
    data class StartResult(
        val accepted: Boolean,
        val code: String,
        val message: String,
    ) {
        companion object {
            @JvmStatic
            fun unavailable(): StartResult = StartResult(
                accepted = false,
                code = "minecraft_verification_reserved",
                message = "Minecraft 世界测试接口已预留，当前尚未开放。",
            )
        }
    }

    companion object {
        const val METHOD_ID = "minecraft"
        const val SESSION_ENDPOINT = "/qo/registration/minecraft/session"
        const val CLAIM_ENDPOINT = "/qo/registration/minecraft/claim"
        const val RESULT_ENDPOINT = "/qo/registration/minecraft/result"
    }
}
